### PR TITLE
CORE-7315 Consensual finality flow e2e tests

### PR DESCRIPTION
**Pull request by lankydan**
Includes `ConsensualFinalityFlow` logic fixes and changes to the flow pipeline to handle closing/erroring sessions that have not yet been initiated.

### CODE REVIEW

File: applications/workers/workers-smoketest/src/smokeTest/kotlin/net/corda/applications/workers/smoketest/ledger/ConsensualLedgerTests.kt [hunk 1]
```diff
@@ -1,63 +1,132 @@
...
 import org.junit.jupiter.api.BeforeAll
-import org.junit.jupiter.api.Disabled
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.api.TestInstance
 import org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS
 import java.util.UUID
 
 /**
- * This file uses the flow tests' CPI, Group Id and X500 names.
+ * This file uses a dedicated CPI, Group Id and X500 names from ConsensualLedgerConfig.
```
Comment (vlajos) [inline applications/workers/workers-smoketest/src/smokeTest/kotlin/net/corda/applications/workers/smoketest/ledger/ConsensualLedgerTests.kt]: The `from...` is not true anymore.

**Comment by lankydan** [comment]
@LWogan I've updated the flow tests to make them compile but might be worth reviewing the test to check that everything is still covered. AFAIK, we should be good, but I don't have the time to check properly.

**Comment by lankydan** [comment]
API change - https://github.com/corda/corda-api/pull/681

File: testing/cpbs/ledger-consensual-demo-app/src/main/kotlin/net/cordapp/demo/consensual/ConsensualDemoFlow.kt [hunk 2]
```diff
@@ -57,34 +56,30 @@ class ConsensualDemoFlow : RPCStartableFlow {
     override fun call(requestBody: RPCRequestData): String {
         log.info("Consensual flow demo starting...")
         try {
-            val request = requestBody.getRequestBodyAs<String>(jsonMarshallingService)
+            val request = requestBody.getRequestBodyAs<InputMessage>(jsonMarshallingService)
 
             val alice = memberLookup.myInfo()
```
Comment (blsemo) [inline testing/cpbs/ledger-consensual-demo-app/src/main/kotlin/net/cordapp/demo/consensual/ConsensualDemoFlow.kt]: Not sure this should still be called `alice`?

**Review by blsemo — COMMENTED** [review]
Generally looks good, just a small niggle about the remainder of hardcoded indentities in the test flow.
Can't comment on the flow engine changes.

**Review by LWogan — COMMENTED** [review]
This looks good to me from a flow session perspective. I don't see a test to error a flow with a session which has not been initiated though? Should we have a smoke test for this?

**Comment by lankydan** [comment]
> This looks good to me from a flow session perspective. I don't see a test to error a flow with a session which has not been initiated though? Should we have a smoke test for this?

Yes, but I don't have time. Could you create a tech debt for the flow worker team for it? @LWogan

**Review by blsemo — DISMISSED** [review]
LGTM (the consensual ledger part)

**Review by owenstanford — APPROVED** [review]

**Review by ronanbrowne — APPROVED** [review]
LGTM - re approving post rebase

**Pull request merged**